### PR TITLE
Fix a couple bugs in ChangeNamespaceService 

### PR DESCRIPTION
**Pull request by genlu**
1. Exclude documents we can't change from referenced documents for change namespace #37521 
2. Fix the ChangeNamespaceService so the reference in the form of member access is handled correctly #37891

### CODE REVIEW

File: src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs [hunk 1]
```diff
@@ -589,11 +588,11 @@ private static async Task<ImmutableArray<ReferencedSymbol>> FindReferencesAsync(
             string newNamespace,
             CancellationToken cancellationToken)
         {
-            // Can't apply change to certain document, simply return unchanged.
+            // Can't apply change to certain document, returns null to indicate this.
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs]: How much work would it be to null annotate this file so we can have this gotcha documented in the signature itself?

File: src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs [hunk 2]
```diff
@@ -589,11 +588,11 @@ private static async Task<ImmutableArray<ReferencedSymbol>> FindReferencesAsync(
...
             // e.g. Razor document (*.g.cs file, not *.cshtml)
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs]: This comment doesn't make sense -- .g.cs files aren't razor files...?

File: src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs (hunk 1 above)
Comment (genlu) [reply src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs]: I will look into this

File: src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs (hunk 2 above)
Comment (genlu) [reply src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs]: removed

File: src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs (hunk 1 above)
Comment (genlu) [reply src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs]: I have enabled nullable for this method only

**Comment by genlu** [comment]
@jasonmalinowski Ping... Could you please take a look?

**Comment by genlu** [comment]
I'm going to use change from #38070 to address #37521

The fix for #37891 is now extracted into a separate PR #38101

**Pull request closed without merging**